src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,150 @@
+                    taskToAdd = new Event(dateAndTask[0], dateTime);
+                } else {
+                    throw new IllegalArgumentException("Please specify type of task");
+                }
+            }
+            // Add task to data file
+            dataToStore = taskToAdd.getDataRep();
+            Files.writeString(dataPath, dataToStore + System.lineSeparator(), StandardOpenOption.APPEND);
+
+            // Add task To arrayList
+            this.tasks.add(taskToAdd);
+
+            return(String.format("Got it. I've added this task:\n" +
+                    "%s\nNumber of tasks: %s", taskToAdd.toString(), tasks.size()));
+
+        } catch (IllegalArgumentException e) {
+            return("OOPS!!! I'm sorry, but I don't know what that means :-(");
+        } catch(IOException | StringIndexOutOfBoundsException | DateTimeParseException  e) {
+            return("Invalid Input format -> <taskType> <task> </by or /at> <yyyy-MM-dd HHmm>");
+        }
+    }
+
+    /**
+     * Deletes a task from the array of tasks.
+     *
+     * @param userInput String of task to delete.
+     */
+    public String deleteTask(String userInput) {
+        try {
+            //Delete from ArrayList
+            int taskToDel = Integer.parseInt(userInput.substring(7)) - 1;
+            Task task = this.tasks.get(taskToDel);
+            this.tasks.remove(taskToDel);
+            return(String.format("Noted. I've removed this task:\n%s\nNow you have %s tasks in list"
+                    , task, this.tasks.size()));
+        } catch (StringIndexOutOfBoundsException e) {
+            return("OOPS!!! You cannot delete nothing!");
+        } catch (NumberFormatException e) {
+            return("OOPS!!! Must be a number bodoh");
+        } catch (IndexOutOfBoundsException e) {
+            return("OOPS!!! Number doesnt exist");
+        }
+    }
+
+    /**
+     * Marks a task as complete.
+     *
+     * @param userInput Text beginning with 'done' followed by a number.
+     */
+    public String markTaskDone(String userInput) {
+        try {
+            int taskIndex = Integer.parseInt(userInput.substring(5)) - 1;
+            Task task = this.tasks.get(taskIndex);
+            if (task.checkStatus()) {
+                return("Youve already marked this task as done!");
+            } else {
+                task.setDone();
+                return("Nice! I've marked this task as done:\n" + task.toString());
+            }
+        } catch (StringIndexOutOfBoundsException e) {
+            return("OOPS!!! You cannot mark nothing as done!");
+        } catch (NumberFormatException e) {
+            return("OOPS!!! Must be a number bodoh");
+        } catch (IndexOutOfBoundsException e) {
+            return("OOPS!!! NUmber doesnt exist");
+        }
+    }
+
+    public List<Task> getTasks() {
+        return this.tasks;
+    }
+
+    /**
+     * Prints the list of todos sequentially.
+     *
```

Comment:
Perhaps the header comments should not have empty lines based on JavaDocs convention?